_posts/2016-06-30-Rails-5-0-final.markdown

```diff
@@ -0,0 +1,83 @@
+
+What's really lovely about Action Cable is that you get access to your entire Active Record and PORO domain model in your WebSockets work. We even added a brand-new [ActionController::Renderer][ac-renderer] system that makes it trivial to render your templates outside of controllers, when you want to reuse server-side templates for WebSocket responses.
+
+In development, Action Cable runs in-process with the rest of your app. To do this, we've switched the default development server from Webrick to [Puma][puma]. In production, you may well want to run Action Cable servers in their own processes. That's how we run it at Basecamp at scale.
+
+Special thanks to [Pratik Naik][pratiknaik], [Javan Makhmali][javanmakhmali] and [Matthew Draper][matthewdraper] for their formative work on Action Cable.
+
+## API mode
+Rails is not only a great choice when you want to build a full-stack application that uses server-side rendering of HTML templates, but also a great companion for the new crop of client-side JavaScript or native applications that just needs the backend to speak JSON. We've made this even clearer now with the new --api mode. If you create a new Rails application using `rails new backend --api`, you'll get a slimmed down skeleton and configuration that assumes you'll be working with JSON, not HTML.
+
+There's still more work to be done on this feature, but we're off to a great start. By default, API mode just relies on `#to_json` calls on model classes. But you can either use [Jbuilder][jbuilder], [Active Model Serializers][ams], or look at the new [JSONAPI::Resources][jsonapi] project for a more advanced solution.
+
+Thanks in particular to [Santiago Pastorino][santiagopastorino] and [Jorge Bejar][jorgebejar] for making this happen.
+
+## Other highlights
+
+* One Rails Command instead of the split-brain setup between rake and rails, so now it's `bin/rails db:migrate` instead of `bin/rake db:migrate` (herding by [Kasper Timm Hansen][kasper]).
+* [New Attributes API][attributes-api] by [Sean Griffin][sean].
+* The test runner now reports failures inline, so you don't have to complete the suite to see what went wrong.
+* ApplicationRecord has been born as a default parent class of all models created by the generators.
+* ActiveRecord::Relation#in_batches makes it much easier to deal with record work in batches at a time to lessen memory overloads.
+* Post.where('id = 1').or(Post.where('id = 2')) gives you exactly what you'd think!
+* No more accidentally halting Active Record callbacks because the last statement is false. Now you throw(:abort) explicitly!
```

Comment:
Yes, but my point is that this sounds like upgrading apps get this new behavior out of the box (which they only do if they run `rails app:update`).

On the other hand, it's probably just fine to leave this to the upgrading guide, since people have to check that anyway 😁